src/components/Insertion/InsertionDisplay/index.tsx

```diff
@@ -1,12 +1,11 @@
 import { collection, query, getDocs, where } from 'firebase/firestore';
 import { db } from '../../../firebaseConfig';
-import { useEffect, useState } from 'react';
 import { InsertionData } from '../types';
 import { useUserAuth } from '../../../context/userAuthContext';
 import MapResults from './MapResults';
+import { useQuery } from '@tanstack/react-query';
 
 const InsertionDisplay: React.FC = () => {
-  const [insertions, setInsertions] = useState<InsertionData[]>([]);
   const user = useUserAuth();
 
   const getData = async () => {
```

Comment:
`getData` la puoi spostare fuori dal componente

come ti dicevo stamattina queste funzioni vengono ricreate ad ogni render, nel tuo caso dipende solo da `user` per l'ID dell'utente, quindi puoi passarlo come argomento della funzione e amen. una roba tipo

```
// prima del componente
function getData(userId: string) {
  // chiamata con filtro usando questo userId
}

// Dentro il componente
useQuery({
  queryFn: () => getData(user.uid)
})

Reply:
Potrebbe andare bene una roba del genere?

  `const getData = async (userId: string) => {
    try {
      const q = query(
        collection(db, 'insertions'),
        where('userId', '==', userId),
      );
      const querySnapshot = await getDocs(q);
      const insertions = querySnapshot.docs.map(doc => {
        const docData = doc.data();
        return {
          id: doc.id,
          ...docData,
        } as InsertionData;
      });

      return insertions;
    } catch (error) {
      console.error('Error fetching documents:', error);
    }
  };

  const { data: insertions, isLoading } = useQuery({
    queryFn: () => getData(user?.uid || ''),
    queryKey: ['insertions'],
  });`

Reply:
Devo aggiungere gli argomenti in queryKey seguendo il discorso di sotto

Reply:
si e si